pandas/core/dtypes/missing.py

```diff
@@ -584,6 +584,8 @@ def _array_equivalent_object(left: np.ndarray, right: np.ndarray, strict_nan: bo
                 if "boolean value of NA is ambiguous" in str(err):
                     return False
                 raise
+            except ValueError:
+                return False
```

Comment:
cc @jbrockmendel are you OK with just generically catching errors from numpy and in that case returning False? 
I would say that if numpy raises, this generally means the arrays can't be compared for some reason, and it can be OK to return False from this method instead of bubbling up that error from numpy (although it depends on what you use this for ..)

Reply:
i dont see any better alternatives.  can you add a comment here, about the relevant cases